pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -12,16 +12,52 @@
 
 from pyccel.ast.datatypes import NativeVoid, NativeGeneric, NativeHomogeneousList
 from pyccel.ast.internals import PyccelInternalFunction
+from pyccel.utilities.stage import PyccelStage
 
+pyccel_stage = PyccelStage()
 
 __all__ = ('ListAppend',
            'ListClear',
+           'ListExtend',
            'ListInsert',
+           'ListMethod',
            'ListPop',
            )
 
 #==============================================================================
-class ListAppend(PyccelInternalFunction):
+class ListMethod(PyccelInternalFunction):
+    """
+    Abstract class for list method calls.
+
+    A subclass of this base class represents calls to a specific list
+    method.
+
+    Parameters
+    ----------
+    list_variable : TypedAstNode
+        The list object which the method is called from.
```

Comment:
if we take this example:
```python
a = [1, 2]
a.append(1)
```
This is what we get if we inspect the args passed to `__init__()` of `ListMethod`:
![argslist](https://github.com/pyccel/pyccel/assets/66710845/63109bf8-023f-4e83-a340-baf3a38545f7)

Reply:
Then why is the type `TypedAstNode`?
What about this example:
```python
class A:
    def __init__(self):
        self.l = [1,2,3]

if __name__ == '__main__':
    a = A()
    a.lappend(4)
```
or this example:
```python
def get_list(a : list[list[int]]):
    return a[0]

a = [[1,2],[3,4]]
get_list(a).append(5)
```

Reply:
my bad, since all possible arguments will be instances of subclasses inherited from `TypedAstNode`, it is then correct to have the general type `TypedAstNode`.
The description could be:
**arg : TypedAstNode
        The object which the method is called from.** 
is that appropriate?

Reply:
Much better

Reply:
fixed 4586b5c562505a1fb6c7aa31fbd10e5f525f39d7

Reply:
@EmilyBourne Do you think this update should apply to all the list method classes that take this argument?

Reply:
Yes